src/server/json_family.cc

```diff
@@ -1060,7 +1060,7 @@ OpResult<long> OpDel(const OpArgs& op_args, string_view key, string_view path,
     const json::Path& path = json_path.AsJsonPath();
     long deletions = json::MutatePath(
         path, [](optional<string_view>, JsonType* val) { return true; }, updater.GetJson(),
-        true /* reverse_traversal */);
```

Comment:
As you can see the fix becomes more and more complicated. This is the right time to revisit the whole approach.

The solution is to introduce a separate method `json::DeletePath(path, updater.GetJson())` without the callback (as you can see here it's NOOP anyways).
in that case `json::DeletePath` should be much simpler because you do not need to iterate in reverse as there is no callback to change upper nodes in-place (you just delete whatever is matched).
In fact, you do not need `nodes_to_mutate` in `json::DeletePath`